src/device_trezor/trezor/protocol.cpp

```diff
@@ -500,12 +551,19 @@ namespace tx {
     tsx_data.set_version(1);
     tsx_data.set_client_version(client_version());
     tsx_data.set_unlock_time(tx.unlock_time);
-    tsx_data.set_num_inputs(static_cast<google::protobuf::uint32>(tx.sources.size()));
+    tsx_data.set_num_inputs(static_cast<google::protobuf::uint32>(input_size));
     tsx_data.set_mixin(static_cast<google::protobuf::uint32>(tx.sources[0].outputs.size() - 1));
     tsx_data.set_account(tx.subaddr_account);
     tsx_data.set_monero_version(std::string(MONERO_VERSION) + "|" + MONERO_VERSION_TAG);
     tsx_data.set_hard_fork(m_aux_data->hard_fork ? m_aux_data->hard_fork.get() : 0);
-    assign_to_repeatable(tsx_data.mutable_minor_indices(), tx.subaddr_indices.begin(), tx.subaddr_indices.end());
+
+    if (client_version() <= 1){
+      assign_to_repeatable(tsx_data.mutable_minor_indices(), tx.subaddr_indices.begin(), tx.subaddr_indices.end());
+    }
+
+    if (tsx_data.hard_fork() >= 13){
```

Comment:
Could use HF_VERSION_CLSAG later when that patch is merged. Also did you mean < ?

Reply:
Sure! I've added TODO note to adapt this piece of code and use `HF_VERSION_CLSAG` once it is available in the code. 

Here we have `>= 13` as it is normal `if`, even though assert would have `<`

Reply:
But CLSAG are implemented from 13. >= 13, they're valid. < 13, they're not. This is erroring out if using CLSAG at 13 or later.

Reply:
Hmmm by this condition I mean "if hard fork version is >= 13" throw an error as CLSAGs are not yet supported by the Trezor and Monero client code. 

Once https://github.com/trezor/trezor-firmware/pull/942 and CLSAG PRs is merged to the Monero I will remove this condition completely. For now it is just simple check to abort transaction which uses too new hard fork which is not yet suported.

Reply:
Oh, I misunderstood the purpose then, sorry.